lib/ddtrace/sampling/rule_sampler.rb

```diff
@@ -50,7 +50,7 @@ def initialize(rules = [],
                              @rules << SimpleRule.new(sample_rate: default_sample_rate)
                              nil
                            else
-                             RateByServiceSampler.new(1.0, env: -> { Datadog.tracer.tags[:env] })
+                             RateByServiceSampler.new(1.0, env: -> { Datadog::Tracing.send(:tracer).tags[:env] })
```

Comment:
I'm curious, why use `Datadog::Tracing.send(:tracer).tags[:env]` rather than `Datadog.configuration.env`? Is there something I'm missing?

Reply:
I don't know why this is reading off the tracer itself. Maybe the tag has been added to the tracer in the format that it prefers, and the configuration is too raw. I was only making a functionally accurate update here.

Reply:
This exists where because `Datadog.tracer.tags` could previously be changed at any point in the program execution.

I don't think that's the case anymore, as we remove `tags=`-like apis in 1.0.

I think we can add a comment here to clean up this tag access.